analyses/bangladesh/trigger_development/bgd-trigger.md

```diff
@@ -0,0 +1,575 @@
+                                sheet_name=None,
+                                header=[1], index_col='Date')
+df_ffwc_wl = (df_ffwc_wl_dict['2017']
+              .append(df_ffwc_wl_dict['2018'])
+                        .append(df_ffwc_wl_dict['2019'])
+                        .rename(columns={**{
+                            f'{leadtime} hrs': leadtime
+                            for leadtime in ffwc_leadtimes
+                        }, **{'Observed WL': 'observed'}}
+                        ))
+# Convert date time to just date
+df_ffwc_wl.index = df_ffwc_wl.index.floor('d')
+
+# Reindex the date column and shift the predictions
+# new_index = pd.date_range(df_ffwc_wl.index.min(), 
+#              df_ffwc_wl.index.max() + np.timedelta64(5, 'D'))
+# df_ffwc_wl = df_ffwc_wl.reindex(new_index)
+
+# for leadtime in ffwc_leadtimes:
+#     df_ffwc_wl[leadtime] = (
+#         df_ffwc_wl[leadtime].shift(int(leadtime/24))
+#     )
+# df_ffwc_wl.dropna(how='all', inplace=True)
+
+# df_ffwc_wl[df_ffwc_wl.index.year==2019]
+```
+
+### Old FFWC data 
+
+- Used by Leonardo for initial trigger dev
+- Goes from 1987-08-12 to 2019-07-29	
+- Has observed daily average, but only when it is above 19.5 m (so there are many gaps)
+
+```python
+FFWC_RL_HIS_FILENAME='2020-06-07 Water level data Bahadurabad Upper danger level.xlsx'
+ffwc_rl_name='{}/{}'.format(ffwc_dir,FFWC_RL_HIS_FILENAME)
+df_ffwc_wl_old=pd.read_excel(ffwc_rl_name,index_col=0,header=0)
+df_ffwc_wl_old.index=pd.to_datetime(df_ffwc_wl_old.index,format='%d/%m/%y')
+df_ffwc_wl_old=df_ffwc_wl_old[['WL']].rename(columns={'WL': 
+                                                    'observed'})
+```
+
+### Other FFWC data
+
+- Hannah obtained this from Hassan
+- observations recorded 5 times a day: 06:00, 09:00, 12:00, 15:00, 18:0
+- other stations also available
+- runs from 2016-01-01 to 2020-10-31
+
+```python
+# Not needed, doesn't contain Bahadarabad
+#ffwc_full_data_filename = "FFWC_data.xls"
+#ffwc_additional_data_filename = "SW46.9L_19-11-2020"
+#df_ffwc_full_wl = pd.read_excel(ffwc_dir / ffwc_full_data_filename, 
+#                                 index_col='DateTime')
+
+
+ffwc_full_data_filename = 'SW46.9L_19-11-2020.xls'
+df_ffwc_full_wl = (pd.read_excel(ffwc_dir / ffwc_full_data_filename, 
+                                 index_col='DateTime')
+                   .rename(columns={'WL(m)': 'observed'}))[['observed']]
+```
+
+### Comparison plot
+
+```python
+fig, ax = plt.subplots()
+ax.plot(df_ffwc_full_wl.index, df_ffwc_full_wl['observed'], '.')
+ax.plot(df_ffwc_wl.index, df_ffwc_wl['observed'], '.')
+ax.plot(df_ffwc_wl_old.index, df_ffwc_wl_old['observed'], '.')
+fig.autofmt_xdate()
+
+ax.set_xlim(df_ffwc_wl.index[0], df_ffwc_wl.index[10])
+
+```
+
+### Full 30 year figure
+
+```python
+fig, ax = plt.subplots(figsize=(20,10))
+ax.plot(df_final.index, df_final['observed'], 'o', 
+        label='FFWC observations')
+ax.plot(df_final.index, df_final['ffwc_5day'], 's', 
```

Comment:
Also getting this... is there something missing from my input ffwc data?

```
KeyError                                  Traceback (most recent call last)
<ipython-input-23-f3e0cbebef3d> in <module>
      2 ax.plot(df_final.index, df_final['observed'], 'o', 
      3         label='FFWC observations')
----> 4 ax.plot(df_final.index, df_final['ffwc_5day'], 's', 
      5         alpha=0.5, label='FFWC 5 day forecast')
      6 idx_event = df_final['event'] == True

~\anaconda3\envs\bang_floods\lib\site-packages\pandas\core\frame.py in __getitem__(self, key)
   2900             if self.columns.nlevels > 1:
   2901                 return self._getitem_multilevel(key)
-> 2902             indexer = self.columns.get_loc(key)
   2903             if is_integer(indexer):
   2904                 indexer = [indexer]

~\anaconda3\envs\bang_floods\lib\site-packages\pandas\core\indexes\base.py in get_loc(self, key, method, tolerance)
   2891                 return self._engine.get_loc(casted_key)
   2892             except KeyError as err:
-> 2893                 raise KeyError(key) from err
   2894 
   2895         if tolerance is not None:

KeyError: 'ffwc_5day'
```

Reply:
Oops good catch! This error came from when I updated to using days instead of hours for leadtimes. I had fixed it in an upstream branch but not here. Should be fixed with the next commit!